pkg/controllers/nodeclaim/lifecycle/initialization.go

```diff
@@ -54,38 +57,73 @@ func (i *Initialization) Reconcile(ctx context.Context, nodeClaim *v1.NodeClaim)
+	if requeue {
+		return reconcile.Result{Requeue: true}, nil
+	}
+	if err != nil {
+		return reconcile.Result{}, err
+	}
 	node, err := nodeclaimutils.NodeForNodeClaim(ctx, i.kubeClient, nodeClaim)
 	if err != nil {
 		nodeClaim.StatusConditions().SetUnknownWithReason(v1.ConditionTypeInitialized, "NodeNotFound", "Node not registered with cluster")
 		return reconcile.Result{}, nil //nolint:nilerr
 	}
 	ctx = log.IntoContext(ctx, log.FromContext(ctx).WithValues("Node", klog.KRef("", node.Name)))
+	if updateNodeClaimInitializedUnknown(node, nodeClaim) {
+		return reconcile.Result{}, nil
+	}
+	stored := node.DeepCopy()
+	node.Labels = lo.Assign(node.Labels, map[string]string{v1.NodeInitializedLabelKey: "true"})
+	if !equality.Semantic.DeepEqual(stored, node) {
+		if err = i.kubeClient.Patch(ctx, node, client.MergeFrom(stored)); err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+	log.FromContext(ctx).WithValues("allocatable", node.Status.Allocatable).Info("initialized nodeclaim")
+	nodeClaim.StatusConditions().SetTrue(v1.ConditionTypeInitialized)
+	return reconcile.Result{}, nil
+}
+
+func updateNodeClaimInitializedUnknown(node *corev1.Node, nodeClaim *v1.NodeClaim) bool {
 	if nodeutils.GetCondition(node, corev1.NodeReady).Status != corev1.ConditionTrue {
 		nodeClaim.StatusConditions().SetUnknownWithReason(v1.ConditionTypeInitialized, "NodeNotReady", "Node status is NotReady")
-		return reconcile.Result{}, nil
+		return true
 	}
 	if taint, ok := StartupTaintsRemoved(node, nodeClaim); !ok {
 		nodeClaim.StatusConditions().SetUnknownWithReason(v1.ConditionTypeInitialized, "StartupTaintsExist", fmt.Sprintf("StartupTaint %q still exists", formatTaint(taint)))
-		return reconcile.Result{}, nil
+		return true
 	}
 	if taint, ok := KnownEphemeralTaintsRemoved(node); !ok {
 		nodeClaim.StatusConditions().SetUnknownWithReason(v1.ConditionTypeInitialized, "KnownEphemeralTaintsExist", fmt.Sprintf("KnownEphemeralTaint %q still exists", formatTaint(taint)))
-		return reconcile.Result{}, nil
+		return true
 	}
 	if name, ok := RequestedResourcesRegistered(node, nodeClaim); !ok {
 		nodeClaim.StatusConditions().SetUnknownWithReason(v1.ConditionTypeInitialized, "ResourceNotRegistered", fmt.Sprintf("Resource %q was requested but not registered", name))
-		return reconcile.Result{}, nil
+		return true
 	}
-	stored := node.DeepCopy()
-	node.Labels = lo.Assign(node.Labels, map[string]string{v1.NodeInitializedLabelKey: "true"})
-	if !equality.Semantic.DeepEqual(stored, node) {
-		if err = i.kubeClient.Patch(ctx, node, client.MergeFrom(stored)); err != nil {
-			return reconcile.Result{}, err
+	return false
+}
+
+func (i *Initialization) updateNodePoolRegistrationHealth(ctx context.Context, nodeClaim *v1.NodeClaim) (bool, error) {
+	nodePool := &v1.NodePool{}
+	if err := i.kubeClient.Get(ctx, types.NamespacedName{Name: nodeClaim.Labels[v1.NodePoolLabelKey]}, nodePool); err != nil {
+		return false, err
```

Comment:
This should probably just return the error and be handled in the reconcile like we do in other reconcile loops.